conventions.md

```diff
@@ -0,0 +1,210 @@
+- **Parameter adjustments:**  
+  After a swap, verify if facet-specific parameters (e.g., expected `outputAmount`) require adjustment based on the actual swap outcome.
+- **Parameter ordering:**  
+  For facets with a `receiverAddress` parameter, it should be the first parameter in the `facetData` struct and must match the `bridgeData.receiver`.
+- **Cross-verification:**  
+  If `facetData` contains a `targetChainId`, verify it against `bridgeData.destinationChain`.
+- **Modifiers and events:**  
+  Ensure usage of default modifiers (e.g., `nonReentrant`, `refundExcessNative`) and that the `LiFiTransferStarted` event is emitted.
+- **Fee handling and non-evm support:**  
+  For native fees, use the `_depositAndSwap` variant that reserves the fee. For non-evm chains (e.g., Bitcoin), ensure the `receiverAddress` is declared as `bytes` (not `bytes32`).
+
+## Solidity tests conventions (.t.sol files)
+
+- **File naming and structure**  
+  - Test files typically have a `.t.sol` extension (e.g., `AcrossFacetV3.t.sol`, `DeBridgeDlnFacet.t.sol`, `DexManagerFacet.t.sol`).  
+  - Each file should begin with the SPDX license identifier and the Solidity version:
+
+        // SPDX-License-Identifier: Unlicense
+        pragma solidity ^0.8.17;
+
+  - Group and order imports with system libraries first and project files next.
+
+- **Test function naming**  
+  - Prefix test functions expected to pass with `test_`.  
+  - Prefix test functions expected to revert with `testRevert_` (using `vm.expectRevert` to check for specific error selectors).  
+  - Use clear and descriptive names that capture the test’s purpose. For example:  
+    1. `test_CanSwapAndBridgeTokensWithOutputAmountPercent`  
+    2. `testRevert_FailsIfCalledWithOutdatedQuote`  
+    3. `test_SucceedsIfOwnerAddsDex`  
+    4. `testRevert_FailsIfNonOwnerTriesToAddDex`  
+  - For base or inherited tests, prefix with `testBase_`.
+
+- **Test structure and setup**  
+  - Every test contract must include a `setUp()` function to initialize the test environment.  
+  - The `setUp()` function typically configures custom block numbers, initializes base contracts, sets up facets, and assigns labels (using `vm.label`) for clarity.  
+  - Common initialization steps include calling `initTestBase()` and setting facet addresses in the test base.  
+  - Use `vm.startPrank(address)` and `vm.stopPrank()` to simulate transactions from different users.
+
+- **Assertions and event testing**  
+  - Use `assertEq()` for checking equality of values (e.g., balances, contract addresses, return values).  
+  - Use custom assertion modifiers such as `assertBalanceChange()` to verify balance changes before and after transactions.  
+  - Before executing a function call that should emit an event, use `vm.expectEmit()` with appropriate flags (to check indexed parameters) and the expected event signature.  
```

Comment:
could be more clear/explicit: 
must always use this version: vm.expectEmit(true, true,true, true, address(facet))
- to make sure that event parameters match
- to make sure event is emitted by the right contract